src/Lib.hs

```diff
@@ -35,8 +39,20 @@ instance Out Block
 epoch :: IO Int
 epoch = round `fmap` getPOSIXTime
 
+-- used to unwrap hex digests that come
+-- from the sha256 function below
+fromJust :: Maybe a -> a
```

Comment:
so, `fromJust` already exists in the `Data.Maybe` package. What you have is nice in that it has a custom error message, but (due to this name clash) I think it might be more clear to get rid of this function, and then hashString could along the lines of:

```
hashString = maybe (error "Something went wrong generating a hash") show . sha256
```